utils/segview.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+'''
+This script allows to load a number of .db or .rdb files and display bits in
+a nice visualization.
+
+When more than one files are loaded, a difference between them is shown.
+Differring bits are highlighted.
+'''
+import argparse
+import re
+import sys
+
+import itertools
+
+# =============================================================================
+
+
+def tagmap(tag):
+    """
+    Maps a specific tag name to its generic name
+    """
+
+    tag = tag.replace("CLBLL_L", "CLB")
+    tag = tag.replace("CLBLL_M", "CLB")
+    tag = tag.replace("CLBLM_L", "CLB")
+    tag = tag.replace("CLBLM_M", "CLB")
+
+    tag = tag.replace("SLICEL", "SLICE")
+    tag = tag.replace("SLICEM", "SLICE")
+
+    tag = tag.replace("LIOB33", "IOB33")
+    tag = tag.replace("RIOB33", "IOB33")
+
+    tag = tag.replace("LIOI3", "IOI3")
+    tag = tag.replace("RIOI3", "IOI3")
+
+    # TODO: Add other tag mappings
+
+    return tag
+
+
+def parse_bit(bit):
+    """
+    Decodes string describing a bit. Returns a tuple (frame, bit, value)
+    """
+    match = re.match("^(!?)([0-9]+)_([0-9]+)$", bit)
+    assert match != None, bit
+
+    val = int(match.group(1) != "!")
+    frm = int(match.group(2))
+    bit = int(match.group(3))
+
+    return frm, bit, val
+
+
+def load_and_sort_segbits(file_name, tagmap=lambda tag: tag):
+    """
+    Loads a segbit file (.db or .rdb). Skips bits containing '<' or '>'
+    """
+
+    # Load segbits
+    segbits = {}
+    with open(file_name, "r") as fp:
+        lines = fp.readlines()
+
+        # Parse lines
+        for line in lines:
+            line = line.strip()
+            fields = line.split()
+
+            if len(fields) < 2:
+                print("Malformed line: '%s'" % line)
```

Comment:
Should this be a terminal error?
If not, should this be output on stderr so it doesn't get lost in the output?